src/compas/geometry/_primitives/cylinder.py

```diff
@@ -0,0 +1,293 @@
+
+    Examples
+    --------
+    >>> from compas.geometry import Plane
+    >>> from compas.geometry import Cylinder
+    >>> plane = Plane([0, 0, 0], [0, 0, 1])
+    >>> cylinder = Cylinder(plane, 5, 7)
+
+    """
+
+    __slots__ = ['_plane', '_radius', '_height']
+
+    def __init__(self, plane, radius, height):
+        self._plane = None
+        self._radius = None
+        self._height = None
+        self.plane = plane
+        self.radius = radius
+        self.height = height
+
+    @classmethod
+    def from_data(cls, data):
+        """Construct a cylinder from its data representation.
+
+        Parameters
+        ----------
+        data : :obj:`dict`
+            The data dictionary.
+
+        Returns
+        -------
+        Cylinder
+            The constructed cylinder.
+
+        Examples
+        --------
+        >>> from compas.geometry import Cylinder
+        >>> from compas.geometry import Plane
+        >>> data = {'plane': Plane.worldXY().data, 'radius': 5., 'height': 7.}
+        >>> cylinder = Cylinder.from_data(data)
+
+        """
+        cylinder = cls(Plane.worldXY(), 1, 1)
+        cylinder.data = data
+        return cylinder
+
+    # ==========================================================================
+    # descriptors
+    # ==========================================================================
+
+    @property
+    def plane(self):
+        """Plane: The plane of the cylinder."""
+        return self._plane
+
+    @plane.setter
+    def plane(self, plane):
+        self._plane = Plane(plane[0], plane[1])
+
+    @property
+    def radius(self):
+        """float: The radius of the cylinder."""
+        return self._radius
+
+    @radius.setter
+    def radius(self, radius):
+        self._radius = float(radius)
+    
+    @property
+    def height(self):
+        """float: The height of the cylinder."""
+        return self._height
+
+    @height.setter
+    def height(self, height):
+        self._height = float(height)
+
+    @property
+    def normal(self):
+        """Vector: The normal of the cylinder."""
+        return self.plane.normal
+
+    @property
+    def diameter(self):
+        """float: The diameter of the cylinder."""
+        return self.radius * 2
+
+    @property
+    def data(self):
+        """Returns the data dictionary that represents the cylinder.
+
+        Returns
+        -------
+        dict
+            The cylinder data.
+
+        """
+        return {'plane': self.plane.data,
+                'radius': self.radius}
+
+    @data.setter
+    def data(self, data):
+        self.plane = Plane.from_data(data['plane'])
+        self.radius = data['radius']
+        self.height = data['height']
+
+    def to_data(self):
+        """Returns the data dictionary that represents the cylinder.
+
+        Returns
+        -------
+        dict
+            The cylinder data.
+
+        """
+        return self.data
+
+    @property
+    def center(self):
```

Comment:
I find `center` here misleading as this is the center of the bottom circle and not the center of the cylinder

Reply:
center will be the actual center of the cylinder as well, as the distance function will be such that it extends half the height both above and below the circle's plane. property center now returns the circle's center though.

Reply:
ah, ok. but then it is strange that the circle is in the center of the cylinder and not represents the base: the bottom of the cylinder.

Reply:
in volumetric modelling, all objects are defined as a distance function centered around the origin. a sphere has the center at the origin, a box extends half of each edge length to either side and likewise, a cylinder extends to top and bottom. it is not an extrusion of a circle. but i see your point, it might not correspond to *expected behavior* of any user. any suggestions?

Reply:
maybe rename to `center_of_mass`? for primitives, this would be easy to calculate of course. for more complex objects a bit less trivial...

Reply:
Ah yes, `center_of_mass` could be a solution